gpjax/fit.py

```diff
@@ -121,48 +115,49 @@ def evaluate(self, model: LinearModel, train_data: gpx.Dataset) -> float:
         # Check inputs.
         _check_model(model)
         _check_train_data(train_data)
-        _check_optim(optim)
-        _check_num_iters(num_iters)
         _check_batch_size(batch_size)
         _check_prng_key(key)
         _check_log_rate(log_rate)
         _check_verbose(verbose)
 
-    # Unconstrained space loss function with stop-gradient rule for non-trainable params.
-    def loss(model: Module, batch: Dataset) -> ScalarFloat:
-        model = model.stop_gradient()
-        return objective(model.constrain(), batch)
-
     # Unconstrained space model.
     model = model.unconstrain()
 
-    # Initialise optimiser state.
-    state = optim.init(model)
+    # needed for OptaxSolver to work
+    if isinstance(solver, jaxopt.OptaxSolver):
+        model = jax.tree_map(lambda x: x.astype(jnp.float64), model)
+
+    # Initialise solver state.
+    solver.fun = _wrap_objective(solver.fun)
+    solver.__post_init__()  # needed to propagate changes to `fun` attribute
+
+    solver_state = solver.init_state(
+        model,
+        get_batch(train_data, batch_size, key) if batch_size != -1 else train_data,
+    )
+    jitted_update = jax.jit(solver.update)
```

Comment:
Actually would not recommend using `jit` here. The body function is traced in the `lax.scan`.